src/sql/src/kafka_util.rs

```diff
@@ -224,3 +224,28 @@ where
     .await
     .map_err(|e| sql_err!("{}", e))?
 }
+
+/// Validates that we can connect to the broker and obtain metadata about the topic.
+pub async fn ensure_topic_exists<C>(
+    consumer: Arc<BaseConsumer<C>>,
+    topic: &str,
+    fetch_metadata_timeout: Duration,
+) -> Result<(), PlanError>
+where
+    C: ConsumerContext + 'static,
+{
+    task::spawn_blocking(|| format!("kafka_ensure_topic_exists:{topic}"), {
+        let topic = topic.to_string();
+        move || {
+            mz_kafka_util::client::get_partitions(
+                consumer.as_ref().client(),
+                &topic,
+                fetch_metadata_timeout,
+            )
+            .map_err(|e| sql_err!("{}", e))?;
```

Comment:
I'm not sure we want to map this directly to a SQL error, do you mind adding a new "Topic Missing" variant to `PlanError`? Reason being, it looks like `get_paritions` can return errors from Kafka itself which maybe could contain sensitive information? 

If we do this conversion using `sql_err!` elsewhere then feel free to ignore me, but this seems like a small thing that could bite us in the future

Reply:
Yeah we do this for all the kafka stuff; we do parse the kafka errors to make better errors, but then just convert those to generic `sql_err`'s. We could make kafka error type variants in `PlanError` if we want, but that would have to be followup